### PR TITLE
Public Cloud: Remove Vault related code

### PR DESCRIPTION
**Pull request by pdostal**
We no longer use Vault so the Azure integration can be removed.

- Related ticket: [poo#108509](https://progress.opensuse.org/issues/108509)
- Verification run: [upload_img@Azure-latest](https://pdostal-server.suse.cz/tests/313), [image_on_AKS](https://pdostal-server.suse.cz/tests/309), [15-SP4-smoketest-Maintenance-Incident](https://pdostal-server.suse.cz/tests/317), [15-SP4-Azure-latest-boottime](https://pdostal-server.suse.cz/tests/310), [Azure-SAP](https://openqa.suse.de/tests/9232558), [JeOS-OpenStack](https://pdostal-server.suse.cz/tests/305)

### CODE REVIEW

File: lib/publiccloud/aws_client.pm [hunk 1]
```diff
@@ -80,9 +53,6 @@ sub init {
             $self->key_id($data->{access_key_id});
             $self->key_secret($data->{secret_access_key});
         }
-    } elsif (!defined($self->key_id) || !defined($self->key_secret)) {
```
Comment (asmorodskyi) [inline lib/publiccloud/aws_client.pm:83]: IMO you need to remove `if` part above also and  you need to change `get_credentials` from get_var to get_required_var

**Comment by asmorodskyi** [comment]
Looks good to me :) 

@ricardobranco777  please review SAP/HA related part and maybe you can provide for Pavel some jobs which he can run to see that SAP/HA will not be broken by this ?

File: lib/publiccloud/aws_client.pm (hunk 1 above)
Comment (pdostal) [reply lib/publiccloud/aws_client.pm:83]: Done 👍

**Review by ricardobranco777 — APPROVED** [review]
LGTM.

You can pick up the 2 jobs for GCE from this [job group](https://openqa.suse.de/tests/overview?groupid=285&build=0115&version=15-SP4&distri=sle) to make sure it doesn't impact SAP/HA.

**Comment by pdostal** [comment]
> You can pick up the 2 jobs for GCE from this [job group](https://openqa.suse.de/tests/overview?groupid=285&build=0115&version=15-SP4&distri=sle) to make sure it doesn't impact SAP/HA.

@ricardobranco777 I am trying with [this test](https://openqa.suse.de/tests/9231528) and failing - please see [my result](https://pdostal-server.suse.cz/tests/318#step/sles4sap/115). I don't think that it is caused by my changes. What do you think?

**Review by jlausuch — APPROVED** [review]
Nice, finally! 
We'll miss you, Vault:)

**Comment by ricardobranco777** [comment]
> > You can pick up the 2 jobs for GCE from this [job group](https://openqa.suse.de/tests/overview?groupid=285&build=0115&version=15-SP4&distri=sle) to make sure it doesn't impact SAP/HA.
> 
> @ricardobranco777 I am trying with [this test](https://openqa.suse.de/tests/9231528) and failing - please see [my result](https://pdostal-server.suse.cz/tests/318#step/sles4sap/115). I don't think that it is caused by my changes. What do you think?

Use the equivalent for Azure.

**Pull request merged**